packages/edit-site/src/components/resizable-frame/index.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useRef } from '@wordpress/element';
+import {
+	ResizableBox,
+	__unstableMotion as motion,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../private-apis';
+import { store as editSiteStore } from '../../store';
+
+// Removes the inline styles in the drag handles.
+const HANDLE_STYLES_OVERRIDE = {
+	position: undefined,
+	userSelect: undefined,
+	cursor: undefined,
+	width: undefined,
+	height: undefined,
+	top: undefined,
+	right: undefined,
+	bottom: undefined,
+	left: undefined,
+};
+
+const MIN_FRAME_SIZE = 340;
+const lerp = ( a, b, amount ) => {
+	return a + ( b - a ) * amount;
+};
+
+function ResizableFrame( { isFull, children } ) {
+	const [ frameSize, setFrameSize ] = useState( {
+		width: '100%',
+		height: '100%',
+	} );
+	const [ isResizing, setIsResizing ] = useState( false );
+	const [ isHovering, setIsHovering ] = useState( false );
+	const [ isOversized, setIsOversized ] = useState( false );
+	const { setCanvasMode } = unlock( useDispatch( editSiteStore ) );
+	const initialAspectRatioRef = useRef( null );
+	const initialComputedWidthRef = useRef( null );
+	const FRAME_TRANSITION = { type: 'tween', duration: isResizing ? 0 : 0.5 };
+
+	// Calculate the frame size based on the window width as its resized.
+	const handleResize = ( event, direction, ref ) => {
+		const updatedWidth = ref.offsetWidth;
+		const updatedHeight = ref.offsetHeight;
+
+		if ( initialComputedWidthRef.current === null ) {
+			initialComputedWidthRef.current = updatedWidth;
+		}
+
+		// Set the initial aspect ratio if it hasn't been set yet.
+		if ( initialAspectRatioRef.current === null ) {
+			initialAspectRatioRef.current = updatedWidth / updatedHeight;
+		}
+
+		// Calculate the intermediate aspect ratio based on the current width.
+		const lerpFactor =
+			1 -
+			Math.max(
+				0,
+				Math.min(
+					1,
+					( updatedWidth - MIN_FRAME_SIZE ) /
+						( 1300 - MIN_FRAME_SIZE )
+				)
+			);
+
+		// Calculate the height based on the intermediate aspect ratio
+		// ensuring the frame arrives at a 9:19.5 aspect ratio.
+		const intermediateAspectRatio = lerp(
+			initialAspectRatioRef.current,
+			9 / 19.5,
+			lerpFactor
+		);
+		const newHeight = updatedWidth / intermediateAspectRatio;
+
+		setIsResizing( true );
+
+		if ( event.clientX < 344 ) {
```

Comment:
Noting for posterity: `clientX` does in fact give us usable numbers even when outside the window (e.g. negative numbers for the left edge).

However, this PR now relies solely on the frame width and the ownerDocument width to determine the `isOversized`  boolean. It's just a tiny bit more generic than relying on `clientX`, but no strong opinion here. I could just as well revert to `clientX`, since it's simple and this `ResizableFrame` component is not really generic/reusable anyway.

Reply:
I think I like the current version better, we can keep as-is